File: customview/src/main/java/dinson/customview/activity/_008_RichEditorActivity.java
package dinson.customview.activity;

import android.os.Bundle;
import android.view.View;

import dinson.customview.R;
import dinson.customview._global.BaseActivity;
import dinson.customview.utils.SoftHideKeyBoardUtil;
import dinson.customview.weight.richeditor.widget.EmojiLayout;
import dinson.customview.weight.richeditor.widget.RichEditor;

import static dinson.customview.R.id.emojiLayout;

public class _008_RichEditorActivity extends BaseActivity implements View.OnClickListener {

    private EmojiLayout mEmojiLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity__008__rich_editor);
        SoftHideKeyBoardUtil.assistActivity(this);

        initUI();
    }

    @Override
    public int setWindowBackgroundColor() {
        return android.R.color.white;
    }

    private void initUI() {
        findViewById(R.id.ib_emoji).setOnClickListener(this);
        findViewById(R.id.cancel).setOnClickListener(this);
        mEmojiLayout = (EmojiLayout) findViewById(emojiLayout);
        RichEditor richEditor = (RichEditor) findViewById(R.id.richEditor);
        mEmojiLayout.setEditTextSmile(richEditor);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.ib_emoji:
                if (mEmojiLayout.getVisibility() == View.VISIBLE) {
                    mEmojiLayout.setVisibility(View.GONE);
                    mEmojiLayout.showKeyboard();
                } else {
                    mEmojiLayout.setVisibility(View.VISIBLE);
                    mEmojiLayout.hideKeyboard();
                }
                break;
            case R.id.cancel:
                onBackPressed();
                break;
        }
    }
}

File: customview/src/main/java/dinson/customview/activity/_008RichEditorActivity.kt
package dinson.customview.activity

import android.os.Bundle
import android.view.View

import dinson.customview.R
import dinson.customview._global.BaseActivity
import dinson.customview.utils.SoftHideKeyBoardUtil
import dinson.customview.weight._008richeditor.widget.EmojiLayout
import dinson.customview.weight._008richeditor.widget.RichEditor

import dinson.customview.R.id.emojiLayout
import kotlinx.android.synthetic.main.activity__008__rich_editor.*

class _008RichEditorActivity : BaseActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity__008__rich_editor)
        SoftHideKeyBoardUtil.assistActivity(this)

        initUI()
    }

    private fun initUI() {
        cancel.setOnClickListener { onBackPressed() }
        ib_emoji.setOnClickListener {
            if (emojiLayout.getVisibility() == View.VISIBLE) {
                emojiLayout.setVisibility(View.GONE);
                emojiLayout.showKeyboard();
            } else {
                emojiLayout.setVisibility(View.VISIBLE);
                emojiLayout.hideKeyboard();
            }
        }
        emojiLayout.setEditTextSmile(richEditor)
    }
}
